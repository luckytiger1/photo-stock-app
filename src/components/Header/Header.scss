.App-header {
  background: black;
}

.favorite-link {
  color: white;
}

@mixin text-style {
  font-family: 'SF UI Display Light';
  font-size: 18px;
  font-weight: 600;
  color: white;
  vertical-align: middle;
  padding-left: 5px;
}

@mixin image-style {
  width: 20px;
  height: 20px;
}

.logo {
  display: flex;
  line-height: 20px;
  padding-left: 50px;
  &-text {
    @include text-style;
  }
  img {
    @include image-style;
  }
}

.favorite,
.search-history,
.search {
  line-height: 20px;
  padding: 0 10px;
  &-text {
    @include text-style;
  }
  img {
    @include image-style;
  }
}

.menu-container {
  padding-right: 30px;

  a::after {
    display: block;
    content: '';
    border-bottom: solid 3px white;
    transform: scaleX(0);
    transition: transform 250ms ease-in-out;
    transform-origin: 0 50%;
  }

  a:hover::after {
    transform: scaleX(1);
    transform-origin: 100% 50%;
  }

  a:hover {
    text-decoration: none !important;
  }
}
