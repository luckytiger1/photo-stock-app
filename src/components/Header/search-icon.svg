<svg width="31" height="31" viewBox="0 0 31 31" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M19.616 18.6924H20.6364L26.1131 24.1949C26.6426 24.7245 26.6426 25.5899 26.1131 26.1195C25.5835 26.6491 24.7181 26.6491 24.1885 26.1195L18.6989 20.6299V19.6095L18.3501 19.2479C16.5418 20.7979 14.0747 21.5987 11.4526 21.1595C7.8618 20.5524 4.9943 17.5558 4.55514 13.9391C3.88347 8.47535 8.4818 3.87702 13.9456 4.54868C17.5622 4.98785 20.5589 7.85535 21.166 11.4462C21.6051 14.0683 20.8043 16.5353 19.2543 18.3437L19.616 18.6924ZM7.07392 12.8799C7.07392 16.0962 9.67017 18.6924 12.8864 18.6924C16.1027 18.6924 18.6989 16.0962 18.6989 12.8799C18.6989 9.66368 16.1027 7.06743 12.8864 7.06743C9.67017 7.06743 7.07392 9.66368 7.07392 12.8799Z" fill="white"/>
</svg>
