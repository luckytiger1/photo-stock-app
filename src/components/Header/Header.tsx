import React from 'react';
import './Header.scss';
import { useMediaQuery } from 'react-responsive';
import { Link, withRouter } from 'react-router-dom';
import logo from './assets/unsplash.svg';
import favoriteIcon from './assets/favorite.svg';
import historyIcon from './assets/history.svg';
import searchIcon from './assets/search-icon.svg';
import NavItem from '../NavItem/NavItem';

interface HeaderProps {
  match: any;
}

export const Header: React.FC<HeaderProps> = ({ match }) => {
  const isMobile = useMediaQuery({ maxWidth: 550 });

  return (
    <header className="App-header d-flex justify-content-between py-4">
      <NavItem
        isMobile={isMobile}
        clsName="logo"
        alt="logo"
        icon={logo}
        label="ImageStock"
      />

      <div className="menu-container d-flex">
        {!match.isExact && (
          <Link to="/">
            <NavItem
              isMobile={isMobile}
              clsName="search"
              alt="search-icon"
              icon={searchIcon}
              label="Search"
            />
          </Link>
        )}
        <Link to="/favorites">
          <NavItem
            isMobile={isMobile}
            clsName="favorite"
            alt="history-icon"
            icon={favoriteIcon}
            label="Favorites"
          />
        </Link>
        <Link to="/history">
          <NavItem
            isMobile={isMobile}
            clsName="search-history"
            alt="history-icon"
            icon={historyIcon}
            label="Search history"
          />
        </Link>
      </div>
    </header>
  );
};

export default withRouter(Header);
