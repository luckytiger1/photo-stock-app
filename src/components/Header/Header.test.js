import React from 'react';
import { shallow } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';
import { Header } from './Header';

beforeEach(() => {
  expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));
});

it('<Header /> renders correctly if match.isExact is true', () => {
  const mockMatch = {
    isExact: true,
  };
  expect(shallow(<Header match={mockMatch} />)).toMatchSnapshot();
});
it('<Header /> renders correctly if match.isExact is false', () => {
  const mockMatch = {
    isExact: false,
  };
  expect(shallow(<Header match={mockMatch} />)).toMatchSnapshot();
});
