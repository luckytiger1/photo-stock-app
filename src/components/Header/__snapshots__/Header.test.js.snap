// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly if match.isExact is false 1`] = `
<header
  className="App-header d-flex justify-content-between py-4"
>
  <NavItem
    alt="logo"
    clsName="logo"
    icon="unsplash.svg"
    isMobile={false}
    label="ImageStock"
  />
  <div
    className="menu-container d-flex"
  >
    <Link
      to="/"
    >
      <NavItem
        alt="search-icon"
        clsName="search"
        icon="search-icon.svg"
        isMobile={false}
        label="Search"
      />
    </Link>
    <Link
      to="/favorites"
    >
      <NavItem
        alt="history-icon"
        clsName="favorite"
        icon="favorite.svg"
        isMobile={false}
        label="Favorites"
      />
    </Link>
    <Link
      to="/history"
    >
      <NavItem
        alt="history-icon"
        clsName="search-history"
        icon="history.svg"
        isMobile={false}
        label="Search history"
      />
    </Link>
  </div>
</header>
`;

exports[`<Header /> renders correctly if match.isExact is true 1`] = `
<header
  className="App-header d-flex justify-content-between py-4"
>
  <NavItem
    alt="logo"
    clsName="logo"
    icon="unsplash.svg"
    isMobile={false}
    label="ImageStock"
  />
  <div
    className="menu-container d-flex"
  >
    <Link
      to="/favorites"
    >
      <NavItem
        alt="history-icon"
        clsName="favorite"
        icon="favorite.svg"
        isMobile={false}
        label="Favorites"
      />
    </Link>
    <Link
      to="/history"
    >
      <NavItem
        alt="history-icon"
        clsName="search-history"
        icon="history.svg"
        isMobile={false}
        label="Search history"
      />
    </Link>
  </div>
</header>
`;
